base/strings/basic.jl

```diff
@@ -29,7 +29,7 @@ convert(::Type{Vector{UInt8}}, s::AbstractString) = bytestring(s).data
 convert(::Type{Array{UInt8}}, s::AbstractString) = bytestring(s).data
 convert(::Type{ByteString}, s::AbstractString) = bytestring(s)
 convert(::Type{Vector{Char}}, s::AbstractString) = collect(s)
-convert(::Type{Symbol}, s::AbstractString) = symbol(s)
+convert(::Type{Symbol}, s::AbstractString) = Symbol(s)
```

Comment:
I guess we should remove this one. Calling the constructor from `convert` doesn't make sense.